File: frontend/src/types/customer.ts
export interface CustomerOut {
  id: number;
  name: string;
  phone: string | null;
  email: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface CustomerCreate {
  name: string;
  phone?: string | null;
  email?: string | null;
  notes?: string | null;
}

export interface CustomerUpdate {
  name?: string;
  phone?: string | null;
  email?: string | null;
  notes?: string | null;
}

File: backend/app/schemas/customer.py
from pydantic import BaseModel, constr
from typing import Optional

class CustomerBase(BaseModel):
    name: constr(min_length=1)
    phone: Optional[str] = None

class CustomerCreate(CustomerBase):
    pass

class CustomerUpdate(CustomerBase):
    pass

class CustomerOut(CustomerBase):
    id: int

    class Config:
        from_attributes = True
